src/components/policy/src/policy/src/update_status_manager.cc

```diff
@@ -233,8 +233,8 @@ UpdateStatusManager::UpdateThreadDelegate::~UpdateThreadDelegate() {
 
 void UpdateStatusManager::UpdateThreadDelegate::threadMain() {
   LOG4CXX_DEBUG(logger_, "UpdateStatusManager thread started (started normal)");
-  sync_primitives::AutoLock auto_lock(state_lock_);
   while (false == stop_flag_) {
+    sync_primitives::AutoLock auto_lock(state_lock_);
```

Comment:
I think this lock protects `stop_flag_` variable. So it should be locked before reading `stop_flag_` value.